File: desktop/BeautyEstiva.Desktop/BeautyEstiva.Desktop/Models/AuthModels.cs
using System.Text.Json.Serialization;

namespace BeautyEstiva.Desktop.Models;

public class LoginRequest
{
    [JsonPropertyName("emailOrUsername")]
    public string EmailOrUsername { get; set; } = string.Empty;

    [JsonPropertyName("password")]
    public string Password { get; set; } = string.Empty;
}

public class LoginResult
{
    [JsonPropertyName("token")]
    public string Token { get; set; } = string.Empty;

    [JsonPropertyName("name")]
    public string Name { get; set; } = string.Empty;

    [JsonPropertyName("surname")]
    public string Surname { get; set; } = string.Empty;

    [JsonPropertyName("email")]
    public string Email { get; set; } = string.Empty;

    [JsonPropertyName("roles")]
    public List<string> Roles { get; set; } = [];
}

public class JwtPayload
{
    [JsonPropertyName("sub")]
    public string Sub { get; set; } = string.Empty;

    [JsonPropertyName("tenantId")]
    public string TenantId { get; set; } = string.Empty;

    [JsonPropertyName("email")]
    public string Email { get; set; } = string.Empty;

    [JsonPropertyName("unique_name")]
    public string UniqueName { get; set; } = string.Empty;

    [JsonPropertyName("role")]
    public object? Role { get; set; }

    [JsonPropertyName("exp")]
    public long Exp { get; set; }
}

public class AuthUser
{
    public string Id { get; set; } = string.Empty;
    public string TenantId { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string Surname { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public List<string> Roles { get; set; } = [];

    public string FullName => $"{Name} {Surname}".Trim();
    public bool IsOwner => Roles.Contains("Owner");
    public bool IsAdmin => Roles.Contains("Admin") || IsOwner;
}

File: flutter/lib/core/models/auth_models.dart
class LoginRequest {
  final String emailOrUsername;
  final String password;

  LoginRequest({required this.emailOrUsername, required this.password});

  Map<String, dynamic> toJson() => {
        'emailOrUsername': emailOrUsername,
        'password': password,
      };
}

class LoginResult {
  final String token;
  final String name;
  final String surname;
  final String email;
  final List<String> roles;

  LoginResult({
    required this.token,
    required this.name,
    required this.surname,
    required this.email,
    required this.roles,
  });

  factory LoginResult.fromJson(Map<String, dynamic> json) {
    return LoginResult(
      token: json['token'] ?? '',
      name: json['name'] ?? '',
      surname: json['surname'] ?? '',
      email: json['email'] ?? '',
      roles: List<String>.from(json['roles'] ?? []),
    );
  }
}

class AuthUser {
  final String id;
  final String tenantId;
  final String name;
  final String surname;
  final String email;
  final List<String> roles;
  final String? profileImageUrl;

  AuthUser({
    required this.id,
    required this.tenantId,
    required this.name,
    required this.surname,
    required this.email,
    required this.roles,
    this.profileImageUrl,
  });

  String get fullName => '$name $surname'.trim();
  String get initials =>
      '${name.isNotEmpty ? name[0] : '?'}${surname.isNotEmpty ? surname[0] : '?'}';
  bool get isOwner => roles.contains('Owner');
  bool get isAdmin => roles.contains('Admin') || isOwner;
  String get displayRole =>
      isOwner ? 'Owner' : (roles.contains('Admin') ? 'Admin' : 'Staff');
}
